zio/detector/file.go

```diff
@@ -158,3 +175,61 @@ func (r *multiFileReader) Close() (err error) {
 	}
 	return
 }
+
+func (r *multiFileReader) NewScanner(ctx context.Context, f filter.Filter, filterExpr ast.BooleanExpr, s nano.Span) (scanner.Scanner, error) {
+	return &multiFileScanner{
+		multiFileReader: r,
+		ctx:             ctx,
+		filter:          f,
+		filterExpr:      filterExpr,
+		span:            s,
+	}, nil
+}
+
+type multiFileScanner struct {
+	*multiFileReader
+	ctx        context.Context
+	filter     filter.Filter
+	filterExpr ast.BooleanExpr
+	span       nano.Span
+
+	mu      sync.Mutex // protects below
```

Comment:
It just occurred to me to ask: Do you expect to have multiple goroutines active in here?

Reply:
That's the wrong question. The right one: Do you expect to have multiple goroutines fiddling with the scanner field?

Reply:
The mutex is to ensure that Stats() calls from the mux:
https://github.com/brimsec/zq/blob/master/driver/driver.go#L43
are synchronized wrt to the scanner change.